src/Bicep.Core/Analyzers/Linter/Rules/BCPL1000.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class BCPL1000 : LinterRule
+    {
+        public BCPL1000() : base(
+            code: "BCPL1000",
+            ruleName: "Parameters Required",
+            description: "A valid template must be parameterized.",
+            enableForCLI: true,
+            enableForEdit: true,
+            diagnosticLevel: Diagnostics.DiagnosticLevel.Error,
+            docUri: "https://bicep/linter/rules/BCPL1000") //TODO: set up online documentation location
```

Comment:
>TODO [](http://example.com/codeflow?start=61&length=4)

We can probably setup hierarchical aka.ms links like aka.ms/bicep/rules/BCPL1000

Reply:
+1, and I think we should probably set this up now rather than leaving placeholder URLs in the code.

You can also use hash-based routing if you want to have a single doc with Markdown bookmarks e.g. `https://aka.ms/bicep/rules#BCPL1000` -> `https://github.com/azure/bicep/docs/linterrules.md#BCPL1000`

Reply:
docUris now look like this:

docUri: "https://aka.ms/bicep/linter/no-hardcoded-env-urls")